GooglePay.PaymentDataCryptography/GoogleKeyProvider.cs

```diff
@@ -63,14 +64,19 @@ public class GoogleKeyProvider : ISignatureKeyProvider
         public async Task<IEnumerable<string>> GetPublicKeys(string protocolVersion)
         {
             await FetchKeysIfNeeded().ConfigureAwait(false);
-            lock (_lock)
+            await _semaphoreSlim.WaitAsync();
```

Comment:
I think this should be `await _semaphoreSlim.WaitAsync().ConfigureAwait(false);` to avoid a deadlock.